### PR TITLE
feat: implement read operations for user service

### LINKED ISSUE #4: Task: Implement Read operations for User service
Read: Add `GET /users`, `GET /users/{id}` endpoints.

### PR DESCRIPTION
**Pull request by acboujida**
closes #4

### CODE REVIEW

File: UserService/Controllers/UserController.cs [hunk 1]
```diff
@@ -30,4 +30,45 @@ public IActionResult CreateUser([FromBody] UserDtoCreate userCreateDto)
...
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            var userDto = _mapper.Map<UserDtoRead>(user);
+            return Ok(userDto);
+        }
+        catch (Exception ex)
```
Comment (yassir-ai) [inline UserService/Controllers/UserController.cs:69]: Maybe you should benefice from the exception throwed by using (printing) the Message from the "ex" exception pbject.

**Review by yassir-ai — COMMENTED** [review]
see comment on individual lines

File: UserService/Controllers/UserController.cs (hunk 1 above)
Comment (acboujida) [reply UserService/Controllers/UserController.cs:69]: I was thinking of leaving that until we start integrating Loki, then we can log exceptions properly.

Comment (yassir-ai) [reply UserService/Controllers/UserController.cs:69]: Sure, it's a good idea, but for debugging stuff, maybe adding the message will help us until we add Loki service

**Pull request merged**